File: src/01-easy/pirates-treasure/solution.lua
local w=tonumber(io.read());local h=tonumber(io.read())
local g={};for y=1,h do g[y]={};local i=1;for v in io.read():gmatch('%d+') do g[y][i]=tonumber(v);i=i+1 end end
local dx={-1,-1,-1,0,0,1,1,1};local dy={-1,0,1,-1,1,-1,0,1}
for y=1,h do for x=1,w do if g[y][x]==0 then
    local ok=true
    for i=1,8 do local nx,ny=x+dx[i],y+dy[i]
        if nx>=1 and nx<=w and ny>=1 and ny<=h and g[ny][nx]~=1 then ok=false end end
    if ok then print((x-1)..' '..(y-1));os.exit() end
end end end

File: src/01-easy/pirates-treasure/solution.hs
import System.IO(hSetBuffering,stdout,BufferMode(NoBuffering))
import System.Exit(exitSuccess)
main::IO()
main=do
    hSetBuffering stdout NoBuffering
    w<-readLn::IO Int;h<-readLn::IO Int
    g<-mapM(\_ ->map read.words<$>getLine)[1..h]::IO[[Int]]
    let dx=[-1,-1,-1,0,0,1,1,1];dy=[-1,0,1,-1,1,-1,0,1]
    sequence_[do
        let ok=all(\i->let nx=x+dx!!i;ny=y+dy!!i in nx<0||nx>=w||ny<0||ny>=h||(g!!ny)!!nx==1)[0..7]
        if ok then putStrLn(show x++" "++show y)>>exitSuccess else return()
        |y<-[0..h-1],x<-[0..w-1],(g!!y)!!x==0]
